### ISSUE TITLE
Stereo-Seq Analysis Pipeline

### ISSUE THREAD

**Issue by TdzBAS** [satijalab/seurat#8969]
Hello together,

Due to the significant increase in the popularity of the stereo-seq platform, I highly recommend implementing wrapper functions for this type of analysis.

I am currently having some difficulty in understanding how to read and analyze it. Do you have any suggestions or ideas? The back-up solution would be using stereopy, which couldn't really convince me.

Cheers!

**Comment by alikhuseynov** [comment]
is there public example dataset output that can be used to make a reader function?

**Comment by DrKaiChen** [comment]
I am also looking forward to Seurat developing a function to load the stereo-seq from BGI similar to Visum and CosMx.

**Comment by dcollins15** [comment]
Thanks for the suggestion!

I can certainly see why a loader would be handy 👍 

In the meantime, assuming that you're able to read your counts and coordinates in from disk, you should be able to do something like:

```R
assay_name = "Stereo"
fov_name = "fov"

object <- CreateSeuratObject(counts, assay = assay_name)

fov <- CreateFOV(
  coordinates, 
  type = "centroids",
  assay = assay_name,
  key = Key(fov_name, quiet = TRUE)
)

fov <- fov[Cells(object)]

object[[fov_name]] <- fov
```

**Comment by DrKaiChen** [comment]
Hi David Collins,

Thank you for providing the information. I am not sure whether we should use the sequencing-based analysis method (e.g., 10 × Visum) or the image-based analysis method (e.g., 10 × Xenium or Nanostring CosMx SMI) using the Seurat. I have already obtained the SAW outputs (https://github.com/STOmics/SAW). Could you please provide some guidance on which method would be approprite on the Stereo-seq (bin50 or cellbin)? If you need any information from me, please let me know. Really appreciate it! 

Thank you, 
Kai

**Comment by alikhuseynov** [comment]
> Hi David Collins,
> 
> Thank you for providing the information. I am not sure whether we should use the sequencing-based analysis method (e.g., 10 × Visum) or the image-based analysis method (e.g., 10 × Xenium or Nanostring CosMx SMI) using the Seurat. I have already obtained the SAW outputs (https://github.com/STOmics/SAW). Could you please provide some guidance on which method would be approprite on the Stereo-seq (bin50 or cellbin)? If you need any information from me, please let me know. Really appreciate it!
> 
> Thank you, Kai

For object construction, if using `CreateFOV` then it is either VisiumV2 class or image-based method. For the analysis itself, I would say since Stereo-seq is sequencing-based method the analysis should be as for VisiumHD.

**Comment by DrKaiChen** [comment]
> > Hi David Collins,
> > Thank you for providing the information. I am not sure whether we should use the sequencing-based analysis method (e.g., 10 × Visum) or the image-based analysis method (e.g., 10 × Xenium or Nanostring CosMx SMI) using the Seurat. I have already obtained the SAW outputs (https://github.com/STOmics/SAW). Could you please provide some guidance on which method would be approprite on the Stereo-seq (bin50 or cellbin)? If you need any information from me, please let me know. Really appreciate it!
> > Thank you, Kai
> 
> For object construction, if using `CreateFOV` then it is either VisiumV2 class or image-based method. For the analysis itself, I would say since Stereo-seq is sequencing-based method the analysis should be as for VisiumHD.

Yes, I agree with you. Stereo-seq should be similar to the Visum HD.

**Comment by DrKaiChen** [comment]
Any example of creating a Seurat object using Stereo-seq data would be very helpful.

**Comment by jeanne09jiang** [comment]
I also had a similar problem. So I put together a pipeline for analyzing Stereo-seq data using Seurat, which works similar to 10x Visium HD. Hope this helps. 
https://github.com/jeanne09jiang/Stereo-seq-2-seurat
